validator/client/runner.go

```diff
@@ -125,18 +126,39 @@ func run(ctx context.Context, v iface.Validator) {
 				continue
 			}
 			performRoles(slotCtx, allRoles, v, slot, &wg, span)
-		case isHealthyAgain := <-healthTracker.HealthUpdates():
-			if isHealthyAgain {
+		case <-v.LastSecondOfSlot():
```

Comment:
i'm kinda worried about contents of this delaying the normal processes of the next slot above. if it happens at the last second of the slot wouldn't that mean we only have 1 second to do the below? otherwise this loop would be delayed.